src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/FluidPump.java

```diff
@@ -57,13 +61,18 @@
     private final int[] outputBorder = { 14, 15, 16, 17, 23, 26, 32, 33, 34, 35 };
 
     private final ItemStack emptyBucket = ItemStackWrapper.wrap(new ItemStack(Material.BUCKET));
+    private final ItemStack emptyBottle = ItemStackWrapper.wrap(new ItemStack(Material.GLASS_BOTTLE));
+    private final ItemStack waterBottle = new ItemStack(Material.POTION);
 
     @ParametersAreNonnullByDefault
     public FluidPump(ItemGroup itemGroup, SlimefunItemStack item, RecipeType recipeType, ItemStack[] recipe) {
         super(itemGroup, item, recipeType, recipe);
 
         addItemHandler(onBreak());
         createPreset(this, this::constructMenu);
+        PotionMeta meta = (PotionMeta) this.waterBottle.getItemMeta();
+        meta.setBasePotionData(new PotionData(PotionType.WATER));
+        this.waterBottle.setItemMeta(meta);
```

Comment:
There isn't really any point in storing this water bottle as a field, when you clone it inside `getFilledBottle` anyway.
You are effectively creating a new instance anyway, so there is absolutely no need to store one in memory all the time.
Just assemble the `ItemStack` within `getFilledBottle` directly.